pytest_jupyterhub/jupyterhub_spawners.py

```diff
@@ -25,3 +29,29 @@ def jupyterhub_spawner_class_config(spawner_class):
 def spawner_config():
     """Allows tests to setup configurations that are specific to the spawner implementation"""
     return Config()
+
+
+@pytest.fixture
+def app_for_spawners(request, jupyterhub_spawner_class_config, spawner_config):
+    """Creates an instance of a JupyterHub application that can be used by different spawner implementation test suites"""
+    config = jupyterhub_spawner_class_config.merge(spawner_config)
```

Comment:
`config` will actually be `None`  here because what [`Config.merge`](https://traitlets.readthedocs.io/en/stable/config-api.html#traitlets.config.Config.merge) does is to merge a dict into the other, but without returning actually anything.

You can verify it by doing:
```python
from traitlets.config import Config
jupyterhub_spawner_class_config = Config({"a": "test"})
spawner_class = Config({"b": "test2"})

config = jupyterhub_spawner_class_config.merge(spawner_config)

print(config) # this will return None
print(jupyterhub_spawner_class_config) #this will return the dict with the two configs merged
```

Let's instead create a reference (like a name alias) of one of the initial dicts and use that instead. More about object references in Python [here](https://realpython.com/lessons/object-references/). So, something like:

```python
final_config = jupyterhub_spawner_class_config
final_config.merge(spawner_config)
```

We could have also used `jupyterhub_spawner_class_config` directly and pass that to the MockHub instance but, it would have hard to guess from the naming that it is holding the merged confing

Reply:
Great! Alright, let me do that instead.